### PR TITLE
Fix funceval of a function returning 16 byte value type.

### PR DESCRIPTION
**Pull request by mikem8361**
Fixes issue #6935.

On Linux and OS X, structs less or equal to 16 bytes (instead of 8 bytes
on Windows) are enregistered. Only 8 bytes is being passed back to
debugger during a funceval of a property or method that returns a 16
byte value type.

To fix this, the 16 byte return value (which is only used for enregistered
structures of that size on xplat) needed to be plumbed from CallTargetWorker
through the MethodDescCallSite macros to the func eval code.

The func eval code needed to also deal with 16 byte results.

NUMBER_RETURNVALUE_SLOTS is the number of ARG_SLOTs that will contain the
maximum enregistered return value.

CordbEval:m_result is now ARG_SLOT[NUMBER_RETURNVALUE_SLOTS].

CallTargetWorker is now passed a pointer and the count of bytes to/of the
return value buffer.

Minor fix to SOS SymbolReader function.

### CODE REVIEW

**Comment by mikem8361** [comment]
@jkotas @noahfalk code review?

/cc @janvorli

File: src/vm/mngstdinterfaces.cpp [hunk 1]
```diff
@@ -228,7 +228,7 @@ LPVOID MngStdItfBase::ForwardCallToManagedView(
             MethodDescCallSite mngItf(pMngItfMD, CRemotingServices::GetStubForInterfaceMethod(pMngItfMD));
 
             // Call the stub with the args we were passed originally.
-            Result = (Object*)mngItf.CallWithValueTypes_RetArgSlot(pArgs);
```
Comment (jkotas) [inline src/vm/mngstdinterfaces.cpp]: Why this change? It does not look right.

Comment (mikem8361) [inline src/vm/mngstdinterfaces.cpp]: CallWithValueTypes_RetArgSlot now takes the pointer and size of the return value because now it is the function used if the return value can be an value type larger than ARG_SLOT.  Call_RetArgSlot is exactly equivalent to the original function before my changes.

Comment (jkotas) [inline src/vm/mngstdinterfaces.cpp]: I see - this comment explains it: `Note "permitvaluetypes" is not really used for anything`

**Comment by mikem8361** [comment]
@jkotas and @noahfalk I'm I good to go (merge the PR)?

**Comment by jkotas** [comment]
LGTM

File: src/ToolBox/SOS/NETCore/SymbolReader.cs [hunk 2]
```diff
@@ -389,23 +389,30 @@ internal static bool GetLocalVariableByIndex(IntPtr symbolReaderHandle, int meth
...
+                return true;
+            }
+            catch
             {
-                Marshal.StructureToPtr(info, ptr, false);
-                ptr = (IntPtr)(ptr.ToInt64() + structSize);
             }
-            return true;
+            return false;
         }
```
Comment (noahfalk) [inline src/ToolBox/SOS/NETCore/SymbolReader.cs]: Does this part of the change relate to 16 byte data type issue? I'm not opposed to it but I couldn't figure out how it related.

**Comment by noahfalk** [comment]
lgtm

File: src/ToolBox/SOS/NETCore/SymbolReader.cs (hunk 2 above)
Comment (mikem8361) [inline src/ToolBox/SOS/NETCore/SymbolReader.cs]: It is unrelated but it makes the FEATURE_GDBJIT support more robust.
